atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/TimestampStateStoreTest.java

```diff
@@ -42,46 +54,116 @@ public void before() {
     }
 
     @Test
-    public void earliestVersionUpdatesWhenAllTimestampsRemovedForVersion() {
-        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 200L), VERSION_1);
-        timestampStateStore.putStartTimestamps(ImmutableSet.of(400L, 800L), VERSION_2);
+    public void cannotPutStartTimestampTwice() {
+        timestampStateStore.putStartTimestamps(ImmutableSet.of(100L, 102L, 104L), VERSION_1);
+        assertThat(timestampStateStore.getStartVersion(102L)).hasValue(VERSION_1);
 
-        assertThat(timestampStateStore.getEarliestLiveSequence()).hasValue(Sequence.of(1L));
+        assertThatThrownBy(() -> timestampStateStore.putStartTimestamps(ImmutableSet.of(101L, 103L, 104L), VERSION_2))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Start timestamp already present in map");
+        timestampStateStore.getStartVersion(101L).ifPresent(value -> assertThat(value)
+                .isEqualTo(VERSION_2));
```

Comment:
That's weird - do we expect this to be present or not? Can we just assert on it either being empty, or being present and having that value?